objects/src/transaction/tx_script.rs

```diff
@@ -1,11 +1,55 @@
 use super::{Digest, Felt, Word};
 use crate::{
     assembly::{Assembler, AssemblyContext, ProgramAst},
+    notes::NoteId,
     utils::collections::{BTreeMap, Vec},
     vm::CodeBlock,
     TransactionScriptError,
 };
 
+// TRANSACTION ARGS
+// ================================================================================================
+
+/// A struct that represents optional transaction arguments.
+///
+/// The [TransactionArgs] object is composed of:
+/// - [struct](TransactionScript): a program that is executed in a transaction after all input
+///   notes have been executed..
+/// - [BTreeMap](NoteArgs): data being put on stack when the note script is executed. Different
+///   to Note Inputs, Note Args can be used by the executing account.
```

Comment:
I don't think we need to specify the types. Something like this will probably be fine:
```Rust
/// Transaction arguments consist of:
/// - transaction script: a program that is executed in a transaction after all input notes
///   scripts have been executed..
/// - Note arguments: data put onto the the stack right before a note script is executed. These
///   are different from note inputs, as the prover can specify arbitrary note args for a note.
```
(also made some wording adjustments)